From dfbce0face626139a7927f8fc79041cdb450cf37 Mon Sep 17 00:00:00 2001
From: ZhongWen <b47623@freescale.com>
Date: Wed, 11 Dec 2013 15:52:49 +0800
Subject: [PATCH 046/115] ENGR00288599 Make Ethernet save data after configure

Add some SharedPreferences to make the Ethernet save data automaticlly after
configured.

Signed-off-by: ZhongWen <b47623@freescale.com>
---
 .../com/fsl/ethernet/EthernetConfigDialog.java     | 13 +++--
 Ethernet/com/fsl/ethernet/EthernetManager.java     | 55 +++++++++++++++++++++-
 Ethernet/com/fsl/ethernet/MainActivity.java        | 27 +++++++++--
 3 files changed, 87 insertions(+), 8 deletions(-)

diff --git a/Ethernet/com/fsl/ethernet/EthernetConfigDialog.java b/Ethernet/com/fsl/ethernet/EthernetConfigDialog.java
index e440ca7..263c99c 100644
--- a/Ethernet/com/fsl/ethernet/EthernetConfigDialog.java
+++ b/Ethernet/com/fsl/ethernet/EthernetConfigDialog.java
@@ -19,6 +19,8 @@ import android.view.View;
 import android.app.AlertDialog;
 import android.content.Context;
 import android.content.DialogInterface;
+import android.content.SharedPreferences;
+import android.content.SharedPreferences.Editor;
 import android.util.Log;
 import android.view.View;
 import android.widget.AdapterView;
@@ -44,7 +46,7 @@ public class EthernetConfigDialog extends AlertDialog implements
     private RadioButton mConTypeManual;
     private EditText mIpaddr;
     private EditText mDns;
-
+    private static String Mode_dhcp = "dhcp";
 
     public EthernetConfigDialog(Context context, EthernetEnabler Enabler) {
         super(context);
@@ -63,7 +65,7 @@ public class EthernetConfigDialog extends AlertDialog implements
         mDns = (EditText)mView.findViewById(R.id.eth_dns_edit);
         if (mEthEnabler.getManager().isConfigured()) {
             EthernetDevInfo info = mEthEnabler.getManager().getSavedConfig();
-            if (info.getConnectMode() == EthernetDevInfo.ETHERNET_CONN_MODE_DHCP) {
+            if (mEthEnabler.getManager().getSharedPreMode().equals(Mode_dhcp)) {
                 mConTypeDhcp.setChecked(true);
                 mConTypeManual.setChecked(false);
                 mIpaddr.setEnabled(false);
@@ -73,8 +75,8 @@ public class EthernetConfigDialog extends AlertDialog implements
                 mConTypeManual.setChecked(true);
                 mIpaddr.setEnabled(true);
                 mDns.setEnabled(true);
-                mIpaddr.setText(info.getIpAddress(), TextView.BufferType.EDITABLE);
-                mDns.setText(info.getDnsAddr(), TextView.BufferType.EDITABLE);
+                mIpaddr.setText(mEthEnabler.getManager().getSharedPreIpAddress(),TextView.BufferType.EDITABLE);
+                mDns.setText(mEthEnabler.getManager().getSharedPreDnsAddress(),TextView.BufferType.EDITABLE);
                 mEthEnabler.getManager().updateDevInfo(info);
                 mEthEnabler.setEthEnabled();
             }
@@ -119,10 +121,13 @@ public class EthernetConfigDialog extends AlertDialog implements
             info.setConnectMode(EthernetDevInfo.ETHERNET_CONN_MODE_DHCP);
             info.setIpAddress(null);
             info.setDnsAddr(null);
+            mEthEnabler.getManager().sharedPreferencesStore(EthernetDevInfo.ETHERNET_CONN_MODE_DHCP, null, null);
         } else {
             info.setConnectMode(EthernetDevInfo.ETHERNET_CONN_MODE_MANUAL);
             info.setIpAddress(mIpaddr.getText().toString());
             info.setDnsAddr(mDns.getText().toString());
+            mEthEnabler.getManager().sharedPreferencesStore(EthernetDevInfo.ETHERNET_CONN_MODE_MANUAL,
+                mIpaddr.getText().toString(), mDns.getText().toString());
         }
         mEthEnabler.getManager().updateDevInfo(info);
         mEthEnabler.setEthEnabled();
diff --git a/Ethernet/com/fsl/ethernet/EthernetManager.java b/Ethernet/com/fsl/ethernet/EthernetManager.java
index cf030ba..6af226f 100644
--- a/Ethernet/com/fsl/ethernet/EthernetManager.java
+++ b/Ethernet/com/fsl/ethernet/EthernetManager.java
@@ -43,6 +43,7 @@ import android.os.Message;
 import android.os.Parcel;
 import android.os.SystemProperties;
 import android.content.SharedPreferences;
+import android.content.SharedPreferences.Editor;
 /**
  * Created by B38613 on 9/27/13.
  */
@@ -96,6 +97,9 @@ public class EthernetManager {
     private INetworkManagementService mNMService;
     private DhcpInfo mDhcpInfo;
     private Handler mTrackerTarget;
+    private String mode;
+    private String ip_address;
+    private String dns_address;
 
     public EthernetManager(Context context) {
         mContext = context;
@@ -118,7 +122,8 @@ public class EthernetManager {
      * @return {@code true} if configured {@code false} otherwise
      */
     public boolean isConfigured() {
-        return "1".equals(SystemProperties.get("net."+ DevName[0] + ".config", "0"));
+        //return "1".equals(SystemProperties.get("net."+ DevName[0] + ".config", "0"));
+        return (getSharedPreMode().equals("manual"))||(getSharedPreMode().equals("dhcp"));
     }
 
     /**
@@ -248,4 +253,52 @@ public class EthernetManager {
         SystemProperties.set("net." + info.getIfName() + ".ip", info.getIpAddress());
     }
 
+    public SharedPreferences sharedPreferences(){
+        SharedPreferences sp = this.mContext.getSharedPreferences("ethernet",
+                Context.MODE_WORLD_WRITEABLE);
+        return sp;
+    }
+
+    public void sharedPreferencesStore(String mode,String ipAddress,String dnsAddress){
+        Editor editor = sharedPreferences().edit();
+        try {
+            editor.putString("conn_mode",mode);
+            editor.putString("mIpaddr",ipAddress);
+            editor.putString("mDns",dnsAddress);
+            } catch (NumberFormatException e) {
+                e.printStackTrace();
+            }
+        editor.commit();
+
+    }
+
+    public String getSharedPreMode(){
+        try {
+            mode = sharedPreferences().getString("conn_mode",null);
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        return mode;
+    }
+
+    public String getSharedPreIpAddress(){
+
+        try {
+            ip_address = sharedPreferences().getString("mIpaddr",null);
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        return ip_address;
+    }
+
+    public String getSharedPreDnsAddress(){
+
+        try {
+            dns_address = sharedPreferences().getString("mDns",null);
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        return dns_address;
+    }
+
 }
diff --git a/Ethernet/com/fsl/ethernet/MainActivity.java b/Ethernet/com/fsl/ethernet/MainActivity.java
index ffc84eb..ec7888a 100644
--- a/Ethernet/com/fsl/ethernet/MainActivity.java
+++ b/Ethernet/com/fsl/ethernet/MainActivity.java
@@ -23,6 +23,9 @@ import android.util.Log;
 import android.view.View;
 import android.widget.TextView;
 import android.widget.Button;
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.content.SharedPreferences.Editor;
 import android.content.DialogInterface;
 import com.fsl.ethernet.EthernetDevInfo;
 import android.view.View.OnClickListener;
@@ -34,11 +37,29 @@ public class MainActivity extends Activity {
     private Button mBtnConfig;
     private Button mBtnCheck;
     private EthernetDevInfo  mSaveConfig;
+    private String TAG = "MainActivity";
+    private static String Mode_dhcp = "dhcp";
+    private boolean shareprefences_flag = false;
+    private boolean first_run = true;
+    public static final String FIRST_RUN = "ethernet";
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.ethernet_configure);
+        SharedPreferences sp = getSharedPreferences("ethernet",
+                Context.MODE_WORLD_WRITEABLE);
+        first_run = sp.getBoolean(FIRST_RUN, false);
+        if (!first_run){
+        Editor editor = sp.edit();
+        try {
+            editor.putBoolean(FIRST_RUN, true);
+            editor.putString("conn_mode",EthernetDevInfo.ETHERNET_CONN_MODE_DHCP);
+            } catch (NumberFormatException e) {
+                e.printStackTrace();
+            }
+            editor.commit();
+        }
         mEthEnabler = new EthernetEnabler(this);
         mEthConfigDialog = new EthernetConfigDialog(this, mEthEnabler);
         mEthEnabler.setConfigDialog(mEthConfigDialog);
@@ -75,9 +96,9 @@ public class MainActivity extends Activity {
                 text.setMovementMethod(ScrollingMovementMethod.getInstance());
                 mSaveConfig = mEthEnabler.getManager().getSavedConfig();
                 if (mSaveConfig != null) {
-                    final String config_detail = "IP address : " + mSaveConfig.getIpAddress() + "\n"
-                            + "DNS address: " + mSaveConfig.getDnsAddr() + "\n"
-                            + "IP mode    : " + mSaveConfig.getConnectMode() + "\n";
+                    final String config_detail = "IP address : " + mEthEnabler.getManager().getSharedPreIpAddress() + "\n"
+                            + "DNS address: " + mEthEnabler.getManager().getSharedPreDnsAddress() + "\n"
+                            + "IP mode    : " + mEthEnabler.getManager().getSharedPreMode() + "\n";
                     text.setText(config_detail);
                 }
             }
-- 
1.8.0

